### PR TITLE
datalad 0.17.4

### PR DESCRIPTION
**Pull request by Porkepix**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by Porkepix** [comment]
Error on Linux suggest Rust is needed for build:

<details>
  <summary>Wheel build error</summary>
  ```
unning build_rust
  
        =============================DEBUG ASSISTANCE=============================
        If you are seeing a compilation error please try the following steps to
        successfully install cryptography:
        1) Upgrade to the latest pip and try again. This will fix errors for most
           users. See: https://pip.pypa.io/en/stable/installing/#upgrading-pip
        2) Read https://cryptography.io/en/latest/installation/ for specific
           instructions for your platform.
        3) Check our frequently asked questions for more information:
           https://cryptography.io/en/latest/faq/
        4) Ensure you have a recent Rust toolchain installed:
           https://cryptography.io/en/latest/installation/#rust
  
        Python: 3.10.6
        platform: Linux-5.15.0-1017-azure-x86_64-with-glibc2.23
        pip: 22.2.2
        setuptools: 63.4.3
        setuptools_rust: 1.5.1
        =============================DEBUG ASSISTANCE=============================
  
    error: can't find Rust compiler
  
    If you are using an outdated pip version, it is possible a prebuilt wheel is available for this package but pip is not able to install from it. Installing from the wheel would avoid the need for a Rust compiler.
  
    To update pip, run:
  
        pip install --upgrade pip
  
    and then retry package installation.
  
    If you did intend to build this package from source, try installing a Rust compiler from your system package manager and ensure it is on the PATH during installation. Alternatively, rustup (available at https://rustup.rs)/ is the recommended way to download and update the Rust compiler toolchain.
  
    This package requires Rust >=1.41.0.
    error: subprocess-exited-with-error
    
    × Building wheel for cryptography (pyproject.toml) did not run successfully.
    │ exit code: 1
    ╰─> See above for output.
    
    note: This error originates from a subprocess, and is likely not a problem with pip.
    full command: /home/linuxbrew/.linuxbrew/Cellar/datalad/0.17.4/libexec/bin/python3.10 /home/linuxbrew/.linuxbrew/Cellar/datalad/0.17.4/libexec/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py build_wheel /tmp/tmp756kvyk4
    cwd: /tmp/datalad--cryptography-20220830-9528-1jy0et3/cryptography-37.0.4
    Building wheel for cryptography (pyproject.toml): finished with status 'error'
    ERROR: Failed building wheel for cryptography
  Failed to build cryptography
  ERROR: Could not build wheels for cryptography, which is required to install pyproject.toml-based projects
```
</details>

However, I don't see why it would be needed only on Linux (mac ARM builds did passed), nor what would have make it needed since the latest release, cryptography doesn't seems new to this one: https://github.com/datalad/datalad/compare/0.17.3...0.17.4

Might it be related to latest gcc/glib changes for Linux?

File: Formula/datalad.rb [hunk 1]
```diff
@@ -44,6 +49,11 @@ class Datalad < Formula
     sha256 "5a3d016c7c547f69d6f81fb0db9449ce888b418b5b9952cc5e6e66843e9dd845"
   end
 
+  resource "cryptography" do
+    url "https://files.pythonhosted.org/packages/89/d9/5fcd312d5cce0b4d7ee8b551a0ea99e4ea9db0fdbf6dd455a19042e3370b/cryptography-37.0.4.tar.gz"
+    sha256 "63f9c17c0e2474ccbebc9302ce2f07b55b3b3fcb211ded18a42d5764f5c10a82"
+  end
```
Comment (chenrui333) [inline Formula/datalad.rb:55]: this needs rust compiler

**Comment by chenrui333** [comment]
> However, I don't see why it would be needed only on Linux (mac ARM builds did passed)

arm builds did not pass, it got skipped due to unbottled dependencies (`git-annex`)

File: Formula/datalad.rb (hunk 1 above)
Comment (Porkepix) [reply Formula/datalad.rb:56]: But it's not in the git diff, so why was it added?

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**